cabal-install/tests/UnitTests/Distribution/Solver/Modular/DSL.hs

```diff
@@ -849,7 +855,18 @@ exResolve
               )
               (exDbPkgs db)
         | otherwise = []
-      targets' = fmap (\p -> NamedPackage (C.mkPackageName p) []) targets
+      targets' =
+        fmap
+          ( \p ->
+              Named
+                ( WithConstraintSource
+                    { constraintInner =
+                        NamedPackage (C.mkPackageName p) []
+                    , constraintSource = ConstraintSourceUnknown
```

Comment:
I think that the `targets` parameter should be replaced with one that contains constraint source information. Right now, the constraint source is unknown because that information is missing. We don't know what this function is getting called with or where the targets come from.